extensions/VisualEditor/ApiVisualEditor.php

```diff
@@ -63,8 +73,12 @@ private function getVRSObject() {
 		// the VRS class to use, defaults to Parsoid
 		$class = 'ParsoidVirtualRESTService';
 		$config = $this->veConfig;
+
 		// the global virtual rest service config object, if any
-		$vrs = $this->getConfig()->get( 'VirtualRestConfig' );
```

Comment:
This ```$this->getConfig()->get( 'VirtualRestConfig' )``` is just a proxy for accessing global variables - so in this particular case it accesses $wgVirtualRestConfig which is defined in newer version of MW in DefaultSettings.php - https://github.com/wikimedia/mediawiki/blob/master/includes/DefaultSettings.php#L7671-L7699
It seems that you could actually totally use it , and then you wouldn't even have to have that getApiSource method at all.
Btw. you can pass domain together with "http://" and it will work ok - https://github.com/wikimedia/mediawiki/blob/master/includes/libs/virtualrest/ParsoidVirtualRESTService.php#L69-L76